### DISCUSSION TITLE
How to load to one relationships automatically?

### DISCUSSION THREAD

**Discussion by andreasvh-conceto** [prisma/prisma#13889] [Q&A] [answered]
Hello,

is there a way -  like a flag i can set,  in order to automatically load all "to 1" relations?  It is pretty hard sometimes to write all the includes and chance is high to forget one. And if i add a to one relation in the future, i have always to adjust the include part of the query. Would be great if it is also recursive. Of course to many relationships should be explicitly marked. I am coming from the java world in JPA it was so helpful, that to one relations were loaded eagerly.

Something like:

```ts
// current situation:
await this.prismaService.orderStatusReport.findUnique({
        where: {
          id: orderStatusReport.id,
        },
        include: {
          purchaseOrder: {
            include: {
              salesOrder: {
                include: {
                  tenant: true,
                  salesOrderPositions: true,
                  status: true,
                  orderCategory: true,
                },
              },
              purchaseOrderPositions: true,
              tenant: true,
              buyerAddress: true,
              supplierAddress: true,
              dispatchPartyAddress: true,
              invoiceRecipientAddress: true,
              ultimateConsigneeAddress: true,
              orderType: true,
            },
          },
          tenant: true,
        },
      });

// what i am looking for:
await this.prismaService.orderStatusReport.findUnique({
        where: {
          id: orderStatusReport.id,
        },
        eagerLoadToOneRelations: true,   //  // for loading to one realations
        eagerLoadToOneRelationsNested: true // for loading to one realations in a nested way.
        include: {
          purchaseOrder: {
            include: {
              salesOrder: {
                include: {
                  salesOrderPositions: true   // still able to load to many relationships if i want
                },
              },
              purchaseOrderPositions: true // still able to load to many relationships if i want
            },
          }
        },
      });
```

Is there something similar in prisma?

Thx and best regards

**Comment by austincrim** [comment, marked as answer]
Hey @andreasvh-conceto 👋,

This feature does not exist in Prisma out-of-the-box. You could maybe build it yourself using [middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware).

One thing I've seen users do is to define a default query include object for a model and spread it into the query options to reduce duplication.

```ts
const defaultUserInclude = {
  posts: true,
  roles: true,
  ...etc 
}

await prisma.user.select({
  where: { id: 1 },
  include: {
    ...defaultUserInclude
  }
})
```

We have a couple of outstanding issues related to the `select` API that you might want to read through or upvote.

https://github.com/prisma/prisma/issues/3796
https://github.com/prisma/prisma/issues/5042/

As always, feel free to open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=)!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
